ddcommon/src/config.rs

```diff
@@ -0,0 +1,32 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2023-Present Datadog, Inc.
+
+pub mod parse_env {
+    use http::Uri;
+    use std::{env, str::FromStr, time::Duration};
+
+    use crate::parse_uri;
+
+    pub fn duration(name: &str) -> Option<Duration> {
+        Some(Duration::from_secs_f32(
+            env::var(name).ok()?.parse::<f32>().ok()?,
+        ))
+    }
+
+    pub fn int<T: FromStr>(name: &str) -> Option<T> {
+        env::var(name).ok()?.parse::<T>().ok()
+    }
+
+    pub fn bool(name: &str) -> Option<bool> {
+        let var = env::var(name).ok()?;
+        Some(var == "true" || var == "1")
```

Comment:
Good question, I'd expect casing to not matter either.

@paullegranddc I believe this is your code, was there a spec you programmed this to?

Reply:
On the other hand, I would maybe push in the other direction and ask us to consider not supporting too much if we can get away with it.

That's because it can create inconsistencies -- if libdatadog starts supporting `FOO=oh_why_yes_please_do_enable_it_my_good_chap`, then everything else at Datadog where we parse environment variables will suddenly become inconsistent, because it won't accept this one as well.

So I dunno... I think at most the function could get extra options as arguments but, yeah... I'd say be careful about how much we support here.

Reply:
I think we should follow what the agent uses for bool parsing: https://pkg.go.dev/strconv (the agent uses [this type casting package](https://github.com/spf13/cast/tree/master), which uses go's strconv)

`ParseBool returns the boolean value represented by the string. It accepts 1, t, T, TRUE, true, True, 0, f, F, FALSE, false, False. Any other value returns an error.`

thoughts?

Reply:
In dd-trace-rb, we only usually accept `true` (case insensitive) and `1`. Hence my comment -- we seem to already have picked up an inconsistency, as `TrUe` works for Ruby but not for the agent, and `t` works for the agent and not for Ruby. :sweat_smile: 

I kinda see this as an eternal arms race, the more we add, the more we need to support so I'd still kinda suggest being conservative and adding as little as possible, but hey, it's just my opinion and I'm unsure if Ruby is ever going to use this helper so :shrug:...

Reply:
> if libdatadog starts supporting FOO=oh_why_yes_please_do_enable_it_my_good_chap, then everything else at Datadog where we parse environment variables will suddenly become inconsistent, because it won't accept this one as well.

I'm thinking of petitioning for `oh_why_yes_please_do_enable_it_my_good_chap` to be the only truthy value we accept in all dd libraries

Reply:
I'm fine with keeping it as is. @morrisonlevi what do you think?

Reply:
I suppose keeping it as-is will work until we need to work on consistency, possibly for remote config?